iceoryx_examples/iceperf/CMakeLists.txt

```diff
@@ -26,49 +26,26 @@ find_package(iceoryx_binding_c CONFIG REQUIRED)
 find_package(iceoryx_hoofs CONFIG REQUIRED)
 
 get_target_property(ICEORYX_CXX_STANDARD iceoryx_posh::iceoryx_posh CXX_STANDARD)
-include(IceoryxPlatform)
-
-add_executable(iceperf-bench-leader main_leader.cpp iceperf_leader.cpp base.cpp iceoryx.cpp iceoryx_c.cpp uds.cpp mq.cpp)
-
-target_link_libraries(iceperf-bench-leader
-    iceoryx_posh::iceoryx_posh
-    iceoryx_binding_c::iceoryx_binding_c
-)
-target_compile_options(iceperf-bench-leader PRIVATE ${ICEORYX_WARNINGS} ${ICEORYX_SANITIZER_FLAGS})
-if(CMAKE_SYSTEM_NAME MATCHES QNX)
-    target_link_libraries(iceperf-bench-leader socket)
-endif()
-
-add_executable(iceperf-bench-follower main_follower.cpp iceperf_follower.cpp base.cpp iceoryx.cpp iceoryx_c.cpp uds.cpp mq.cpp)
 
-target_link_libraries(iceperf-bench-follower
-    iceoryx_posh::iceoryx_posh
-    iceoryx_binding_c::iceoryx_binding_c
-)
-target_compile_options(iceperf-bench-follower PRIVATE ${ICEORYX_WARNINGS} ${ICEORYX_SANITIZER_FLAGS})
-if(CMAKE_SYSTEM_NAME MATCHES QNX)
-    target_link_libraries(iceperf-bench-follower socket)
-endif()
-
-add_executable(iceperf-roudi
-    roudi_main_static_config.cpp
-)
+include(IceoryxPlatform)
+include(IceoryxPackageHelper)
 
-target_link_libraries(iceperf-roudi
-    PRIVATE
-    iceoryx_hoofs::iceoryx_hoofs
-    iceoryx_posh::iceoryx_posh_roudi
+iox_add_executable(
+    TARGET      iceperf-bench-leader
+    FILES       main_leader.cpp iceperf_leader.cpp base.cpp iceoryx.cpp iceoryx_c.cpp uds.cpp mq.cpp
+    LIBS        iceoryx_posh::iceoryx_posh iceoryx_binding_c::iceoryx_binding_c
```

Comment:
This is not working here since we also want to access functionality from within `iceoryx_posh` directly. With transitive dependencies we can link against methods from within `iceoryx_posh` but cannot use them with a separate include.